core/modules/xrdsvc/StreamBuffer.cc

```diff
@@ -0,0 +1,94 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "xrdsvc/StreamBuffer.h"
+
+// Third-party headers
+#include "boost/utility.hpp"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.xrdsvc.StreamBuffer");
+}
+
+
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
+
+std::atomic<size_t> StreamBuffer::_totalBytes(0);
+
+// Factory function, because this should be able to delete itself when Recycle() is called.
+StreamBuffer::Ptr StreamBuffer::createWithMove(std::string &input) {
+     Ptr ptr(new StreamBuffer(input));
+     ptr->_selfKeepAlive = ptr;
```

Comment:
Oh. I see what you did there. Is it xrootd forcing you do this?

Reply:
It's either do the move or it has to be copied into some form of buffer later on. I don't really like the idea of passing around a char[], so I just tried to keep the weirdness confined to StreamBuffer.